pkg/proxy/backend/backend.go

```diff
@@ -0,0 +1,67 @@
+package backend
+
+import (
+	"context"
+	"log/slog"
+	"net/url"
+
+	"net/http/httputil"
+
+	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	proxyv1 "github.com/rancher/opni/pkg/apis/proxy/v1"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+type Backend interface {
+	RewriteProxyRequest(string, *corev1.ReferenceList) (func(*httputil.ProxyRequest), error)
+}
+
+type implBackend struct {
+	backendURL   string
+	pluginClient proxyv1.RegisterProxyClient
+	logger       *slog.Logger
+}
+
+func NewBackend(logger *slog.Logger, client proxyv1.RegisterProxyClient) (Backend, error) {
```

Comment:
If this backend is always intended to be used inside a plugin, the constructor and struct should accept a context so that it can inherit the plugin's context

Reply:
The backend works the other way around.  It takes in a plugin client which it uses to call down to the plugin to construct the auth headers to inject in the request to the backend.